src/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -398,11 +404,15 @@ private void populateDirectoryList() {
         if (mFile != null) {
             // TODO Enable when "On Device" is recovered ?
             Vector<OCFile> files = getStorageManager().getFolderContent(mFile/*, false*/);
-            List<HashMap<String, OCFile>> data = new LinkedList<>();
+            files = sortFileList(files);
+
+            List<HashMap<String, Object>> data = new LinkedList<>();
             for (OCFile f : files) {
-                HashMap<String, OCFile> h = new HashMap<>();
+                if (f.isFolder()) {
```

Comment:
This leads to a regression. In master the app is now showing also the files, not just the folders. The ```if``` sentence should be removed.